contracts/VeERC20.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+
+import '@openzeppelin/contracts/utils/Context.sol';
+
+import './interfaces/IVeERC20.sol';
+
+/// @title VeERC20
+/// @notice Modified version of ERC20 where transfers and allowances are disabled.
+/// @dev Only minting and burning are allowed. The hook `_beforeTokenOperation` and
+/// `_afterTokenOperation` methods are called before and after minting/burning respectively.
+contract VeERC20 is Context, IVeERC20 {
```

Comment:
note: this is a modified version of https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol

this is similar to what PTP does in https://snowtrace.io/address/0x0104eC62Afc47aF38CE214568927287E4bfDc773#code